Test/git-issues/git-issue-3804.dfy

```diff
@@ -0,0 +1,64 @@
+// RUN: %exits-with 4 %baredafny verify %args "%s" > "%t"
+// RUN: %diff "%s.expect" "%t"
+
+predicate P(x: int)
+
+function FirstAssertNotProved(): (i: int)
+  ensures P(i)
+{
+  assert rule2: P(12) by {
+  }
+  assert P(12) by {
+    reveal rule2;
+  }
+  12
+}
+
+function BothAssertNotProved(): (i: int)
+  ensures P(i)
+{
+  assert rule2: P(12) by {
```

Comment:
I expect this to fail as well. That is, the verifier should try to prove the asserted condition, whether or not there is a label. But if there is a label, then the asserted condition should be available downstream only if a `reveal` is used.